src/main/java/com/hierynomus/smbj/share/Share.java

```diff
@@ -260,12 +262,55 @@ public int ioctl(int ctlCode, boolean isFsCtl, byte[] inData, int inOffset, int
     }
 
     int ioctl(SMB2FileId fileId, int ctlCode, boolean isFsCtl, byte[] inData, int inOffset, int inLength, byte[] outData, int outOffset, int outLength) {
```

Comment:
Wrong place for this logic. The BUFFER_OVERFLOW logic is only for NamedPipe, not for all other ioctl ops. Move this into `NamedPipe.ioctl`